lib/codebreaker/game_controller.rb

```diff
@@ -0,0 +1,111 @@
+require_relative './game_model.rb'
+
+ATTEMPTS_COUNT = 10
+CODE_LENGTH = 4
+
+module Codebreaker
+  class Game
+    attr_reader :attemps_count, :hint_array, :secret_code
+
+    def initialize
+      @model = Codebreaker::GameModel.new
+    end
+
+    def start
+      puts 'The game has begun'
+      default_settings
+      generate_secret_code
+    end
+
+    def guess(code)
+      @attemps_count -= 1
+
+      if @secret_code == code
+        @player_win = true
+        return
+      end
+
+      if valid_secret_code?(code)
+        add_hint(guessed_hint(@secret_code, code)) if @attemps_count != 0
+      else
+        raise 'Invalid code'
+      end
+
+      puts "#{@attemps_count} out of #{ATTEMPTS_COUNT} attempts left"
+    end
+
+    def reset
+      default_settings
+      generate_secret_code
+    end
+
+    def save(name)
+      @model.save_score(name, @attemps_count)
+    end
+
+    def game_process
+      if @player_win
+        :win
+      elsif @attemps_count.zero? && !@player_win
+        :lose
+      else
+        :play
+      end
+    end
+
+    def max_attempts_count
+      ATTEMPTS_COUNT
+    end
+
+    def hint
+      random_index = rand(CODE_LENGTH)
+      @secret_code[random_index]
+    end
+
+    def statistics
+      @model.load_score
+    end
+
+    private
+
+    def valid_secret_code?(secret_code)
```

Comment:
Can we rewrite this method in one line (without explicitly returning `true` or `false`)?

`secret_code.length < CODE_LENGTH || secret_code.length > CODE_LENGTH`